### PR TITLE
Add javadoc to clear up implicit behavior

### LINKED ISSUE #3982: Constructors that accept a URI and a JedisClientConfig pull elements from both without documentation
UnifiedJedis and possibly other type constructors that accept a `URI` and a `JedisClientConfig` instantiate themselves with a mix of elements from both. This isn't unreasonable given that the URI can contain more than just host and port, but it isn't documented.

Options to fix it:

* Document the current behavior
* Update the constructors to let, for example, any config options win if both are specified, and document the behavior.

For example:

```java
// works
UnifiedJedis jedis = new UnifiedJedis(new HostAndPort("localhost", 6379), DefaultJedisClientConfig.builder().user("username").password("password").build());

// also works
UnifiedJedis jedis2 = new UnifiedJedis(URI.create("redis://username:password@localhost:6379"));

// doesn't set auth correctly
UnifiedJedis ohNo = new UnifiedJedis(URI.create("redis://localhost:6379"), DefaultJedisClientConfig.builder().user("username").password("password").build());
```

See https://github.com/redis/jedis/blob/54426424a21070bba3b12454e48aa692efc3a9fa/src/main/java/redis/clients/jedis/UnifiedJedis.java#L90

### PR DESCRIPTION
**Pull request by joshrotenberg**
This should probably be fixed with some kind of configuration merging, but in an effort to avoid breaking changes, these docs should at least make the behavior clear.

Fixes #3982.

### CODE REVIEW

File: src/main/java/redis/clients/jedis/Jedis.java [hunk 1]
```diff
@@ -193,6 +193,18 @@ public Jedis(final URI uri, final int connectionTimeout, final int soTimeout,
...
   }
 
+  /**
+   * Create a new Jedis with the provided URI and JedisClientConfig object. Note that all fields
+   * that can be parsed from the URI will be used instead of the corresponding configuration values. This includes
+   * the following fields: user, password, database, protocol version, and whether to use SSL.
+   *
+   * For example, if the URI is "redis://user:password@localhost:6379/1", the user and password fields will be set
+   * to "user" and "password" respectively, the database field will be set to 1. If those fields are also set in
+   * the JedisClientConfig object, the URI values will take precedence.
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Jedis.java:203]:
```suggestion
   * the JedisClientConfig object, the URI values will take precedence even if they are absent in URI.
```

File: src/main/java/redis/clients/jedis/UnifiedJedis.java [hunk 2]
```diff
@@ -82,6 +82,18 @@ public UnifiedJedis(final URI uri) {
...
   }
 
+  /**
+   * Create a new UnifiedJedis with the provided URI and JedisClientConfig object. Note that all fields
+   * that can be parsed from the URI will be used instead of the corresponding configuration values. This includes
+   * the following fields: user, password, database, protocol version, and whether to use SSL.
+   *
+   * For example, if the URI is "redis://user:password@localhost:6379/1", the user and password fields will be set
+   * to "user" and "password" respectively, the database field will be set to 1. If those fields are also set in
+   * the JedisClientConfig object, the URI values will take precedence.
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/UnifiedJedis.java:92]:
```suggestion
   * the JedisClientConfig object, the URI values will take precedence even if they are absent in URI.
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]
@joshrotenberg Thank you for this PR. Suggested some changes, please address. You can your own words in sentences.

File: src/main/java/redis/clients/jedis/Jedis.java [hunk 3]
```diff
@@ -193,6 +193,18 @@ public Jedis(final URI uri, final int connectionTimeout, final int soTimeout,
         .hostnameVerifier(hostnameVerifier).build());
   }
 
+  /**
+   * Create a new Jedis with the provided URI and JedisClientConfig object. Note that all fields
+   * that can be parsed from the URI will be used instead of the corresponding configuration values. This includes
+   * the following fields: user, password, database, protocol version, and whether to use SSL.
+   *
+   * For example, if the URI is "redis://user:password@localhost:6379/1", the user and password fields will be set
```
[the remark this answers is not in the document]
Comment (joshrotenberg) [reply src/main/java/redis/clients/jedis/Jedis.java:201]: Not sure what you're wanting changed here. That extra colon is wrong ... did you want the SSL scheme in the URI (`rediss`)

**Comment by joshrotenberg** [comment]
I made a few changes, let me know what you think.

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**